app/api.py

```diff
@@ -297,6 +301,37 @@ def get(self):
             self, ApiActionLog.READ, len(records), len(notes))
 
 
+class PhotoUpload(utils.BaseHandler):
+    https_required = True
+
+    def post(self):
+        if not (self.auth):
```

Comment:
I believe we should allow this API only for people with "write" access. Can you require self.auth and self.auth.domain_write_permission, in the same way as Write class?